### PR TITLE
Added feature for full screen

### PR DESCRIPTION
**Pull request by RigvedRocks**
**What does this PR do?**
I have added the functionality for the full-screen feature as outlined in #202 

**Changes**
I have added the full-screen feature to the project. In this I have made modifications to the following files and folders - 
i) Added two new files in the Icon folder for full-screen and exit full-screen icons - FullScreen.tsx and ExitFullScreen.tsx
ii) Made changes to the HeaderTimer.tsx
iii) Create a new file named fullscreenStore.tsx in the store folder.

**Screenshots or GIFs (if applicable)**
Screenshot (101)
Screenshot (102)

**By submitting this PR, I confirm that:**
-  I have reviewed my code and believe it is ready for merging.
-  I understand that this PR may be subject to review and changes.
-  I agree to abide by the code of conduct and contributing guidelines of this project.

### CODE REVIEW

**Review by bryanlundberg — COMMENTED** [review]
Hi @RigvedRocks  

The overall implementation for the fullscreen logic works as I expected, but I have some issues with the UI.

Could not be merged it in its current state because you made changes in an old fork. I would suggest updating the repository before continuing your work on it.

Additionally, please switch to icons from `heroicons` and avoid committing files without any changes. Thank you.

**Comment by RigvedRocks** [comment]
Thank you so much for the insights, bryanlundberg. Based on your suggestions, I have modified the code. Please have a look at the latest commit and suggest amendments if any.
Screenshot (104)
Screenshot (105)

File: src/components/timer/HeaderTimer.tsx [hunk 1]
```diff
@@ -44,6 +48,11 @@ export default function HeaderTimer() {
             }
           }}
         />
+
+        <InteractiveIcon
+          icon={isFullScreen ? <ExitFullScreen /> : <FullScreen />}
+          handleClick={toggleFullScreen}
+        />
```
Comment (bryanlundberg) [inline src/components/timer/HeaderTimer.tsx:55]: I'm concerned about the placement of this icon. As the repository hasn't been updated with the latest merge, this section might have an unfamiliar appearance.

**Review by bryanlundberg — CHANGES_REQUESTED** [review]
I have to say, very nice work i like it! @RigvedRocks 

I would like to merge just the hook and icons for now. It is hard for me to determine the icon position of this feature yet.

File: src/components/timer/HeaderTimer.tsx (hunk 1 above)
Comment (RigvedRocks) [reply src/components/timer/HeaderTimer.tsx:57]: Thank you so much for the feedback. Can you shed some more light on the placement of the icon?

Comment (RigvedRocks) [reply src/components/timer/HeaderTimer.tsx:57]: Okay, I will just merge the hook and icons.

**Comment by RigvedRocks** [comment]
Hey @bryanlundberg. Can you please guide me as to how I can just merge the hook and icons. I am new to the field of open-source development, I will be happy if you can help me. Sorry for the trouble.

**Comment by RigvedRocks** [comment]
@bryanlundberg Can you please guide me as to how I can just merge the hooks and icons. I am new to the field of open-source development. Sorry for the trouble. If you want to I can close this draft PR, and open a new one with the hooks and icons.

File: src/components/timer/HeaderTimer.tsx [hunk 2]
```diff
@@ -16,6 +19,9 @@ export default function HeaderTimer() {
     useSettingsModalStore();
   const { global } = useTimerStatistics();
   const { lastSolve } = useTimerStore();
+  const { isFullScreen, toggleFullScreen } = useFullScreen();
+  
+  if (isSolving || timerStatus === "ready") return null;
 
   if (isSolving || timerStatus !== "IDLE") return null;
```
Comment (bryanlundberg) [inline src/components/timer/HeaderTimer.tsx:26]: This is not correctly merged

**Review by bryanlundberg — COMMENTED** [review]
If you prefer, you can simply create a new pull request with the hook and icons, it might make things easier. 😅

Additionally, I have some other ideas about where to position the fullscreen button.

File: src/components/timer/HeaderTimer.tsx (hunk 2 above)
Comment (RigvedRocks) [reply src/components/timer/HeaderTimer.tsx:26]: Okay, I will see to it, thank for pointing it out

**Comment by RigvedRocks** [comment]
Closing this draft PR and opening a new one, merging the hook and icons along with changes to the timer component.

**Pull request closed without merging**